examples/dynamic_graph/test_dynamic_graph.cu

```diff
@@ -0,0 +1,97 @@
+// ----------------------------------------------------------------
+// Gunrock -- Fast and Efficient GPU Graph Library
+// ----------------------------------------------------------------
+// This source code is distributed under the terms of LICENSE.TXT
+// in the root directory of this source distribution.
+// ----------------------------------------------------------------
+
+/**
+ * @file
+ * test_dynamic_graph.cu
+ *
+ * @brief Simple test driver program for dynamic graph building.
+ */
+
+#include <gunrock/gunrock.h>
+
+
+// Utilities and correctness-checking
+#include <gunrock/util/test_utils.cuh>
+
+// Graph defintions
+#include <gunrock/graphio/graphio.cuh>
+#include <gunrock/app/app_base.cuh>
+
+#include <gunrock/app/test_base.cuh>
+
+using namespace gunrock;
+
+/******************************************************************************
+ * Main
+ ******************************************************************************/
+
+/**
+ * @brief Enclosure to the main function
+ */
+struct main_struct {
+  /**
+   * @brief the actual main function, after type switching
+   * @tparam VertexT    Type of vertex identifier
+   * @tparam SizeT      Type of graph size, i.e. type of edge identifier
+   * @tparam ValueT     Type of edge values
+   * @param  parameters Command line parameters
+   * @param  v,s,val    Place holders for type deduction
+   * \return cudaError_t error message(s), if any
+   */
+  template <typename VertexT,  // Use int as the vertex identifier
+            typename SizeT,    // Use int as the graph size type
+            typename ValueT>   // Use int as the value type
+  cudaError_t
+  operator()(util::Parameters &parameters, VertexT v, SizeT s, ValueT val) {
+    
+    using WeightedGraphT = app::TestGraph<VertexT, SizeT, ValueT,
+                                    graph::HAS_EDGE_VALUES | 
+                                    graph::HAS_CSR | 
+                                    graph::HAS_COO |
+                                    graph::HAS_DYN>;
+
+    using CSRGraphT = app::TestGraph<VertexT, SizeT, ValueT,
+                                    graph::HAS_EDGE_VALUES | 
+                                    graph::HAS_CSR>;
```

Comment:
Do we need `using CSRGraphT` for some reason?

Reply:
I will remove this line. 
I was doing a complete conversion here but I moved everything to the google test file. This file will eventually contain an example to insertion/deletion.

Reply:
Fixed in https://github.com/gunrock/gunrock/pull/713/commits/18776ca4aed356a2fd692983e006847e5496ebe1.